docs/tasks/administer-cluster/guaranteed-scheduling-critical-addon-pods.md

```diff
@@ -35,18 +41,27 @@ while the other pods shouldn't tolerate the taint. The taint is removed once the
 
 *Warning:* currently there is no guarantee which node is chosen and which pods are being killed
 in order to schedule critical pods, so if rescheduler is enabled your pods might be occasionally
-killed for this purpose.
+killed for this purpose. Please ensure that rescheduler is enabled along with priorities & preemptions in default-scheduler as it may have unwanted side-effects.
 
 ## Config
 
-Rescheduler should be enabled by default. It doesn't have any user facing configuration (component config) or API.
-Rescheduler may also be disabled.
+Rescheduler doesn't have any user facing configuration (component config) or API.
 
-### Marking add-on as critical
+### Marking pod as critical when using Rescheduler(This will be deprecated).
 
-To be critical an add-on has to run in `kube-system` namespace (configurable via flag) and
+To be considered critical, the pod has to run in the `kube-system` namespace (configurable via flag) and
 
 * have the `scheduler.alpha.kubernetes.io/critical-pod` annotation set to empty string, and
-* have the PodSpec's `tolerations` field set to `[{"key":"CriticalAddonsOnly", "operator":"Exists"}]`
+* have the PodSpec's `tolerations` field set to `[{"key":"CriticalAddonsOnly", "operator":"Exists"}]`.
 
 The first one marks a pod a critical. The second one is required by Rescheduler algorithm.
+
+### Marking pod as critical when priorites are enabled.
+
+To be considered critical, the pod has to run in the `kube-system` namespace (configurable via flag) and
+
+* have the priorityClass set as "system-cluster-critical" or "system-node-critical", the latter being the highest for entire cluster.
+
+or
```

Comment:
I think we should remove this part and add that you can use both priority and the critical annotation at the same time.

Reply:
Ack.